encrypted-media/scripts/playback-temporary.js

```diff
@@ -0,0 +1,155 @@
+            _playingEvent = false,
+            _timeupdateEvent = false,
+            _events = [ ];
+
+        function onMessage(event) {
+            assert_equals( event.target, _mediaKeySession );
+            assert_true( event instanceof window.MediaKeyMessageEvent );
+            assert_equals( event.type, 'message');
+
+            assert_any( assert_equals,
+                        event.messageType,
+                        [ 'license-request', 'individualization-request' ] );
+
+            if ( event.messageType !== 'individualization-request' ) {
+                _events.push( event.messageType );
+            }
+
+            config.messagehandler( config.keysystem, event.messageType, event.message ).then( function( response ) {
+                _events.push( 'license-response' );
+
+                waitForEventAndRunStep('keystatuseschange', _mediaKeySession, onKeyStatusesChange, test);
+                _mediaKeySession.update( response ).then( function() {
+                    _events.push('updated');
+                }).catch(function(error) {
+                    forceTestFailureFromPromise(test, error);
+                });
+            });
+        }
+
+        function onKeyStatusesChange(event) {
+            assert_equals(event.target, _mediaKeySession );
+            assert_true(event instanceof window.Event );
+            assert_equals(event.type, 'keystatuseschange' );
+
+            var hasKeys = false, pendingKeys = false;
+            _mediaKeySession.keyStatuses.forEach( function( value, keyid ) {
+                assert_any( assert_equals, value, [ 'status-pending', 'usable' ] );
+
+                hasKeys = true;
+                pendingKeys = pendingKeys || ( value === 'status-pending' );
+
+            });
+
+            if ( !_allKeysUsableEvent && hasKeys && !pendingKeys ) {
+                _allKeysUsableEvent = true;
+                _events.push( 'allkeysusable' );
+                _video.setMediaKeys(_mediaKeys);
+            }
+
+            if ( !hasKeys ) {
+                _events.push( 'emptykeyslist' );
+            }
+        }
+
+        function onEncrypted(event) {
+            assert_equals(event.target, _video);
+            assert_true(event instanceof window.MediaEncryptedEvent);
+            assert_equals(event.type, 'encrypted');
+
+            waitForEventAndRunStep('message', _mediaKeySession, onMessage, test);
+            _mediaKeySession.generateRequest(   config.initDataType || event.initDataType,
+                                                config.initData || event.initData ).then( function() {
+                _events.push( 'generaterequest' );
+            }).catch(function(error) {
+                forceTestFailureFromPromise(test, error);
+            });
+        }
+
+        function onClosed(event) {
+
+            _events.push( 'closed-promise' );
+
+            setTimeout( test.step_func( function() {
+
+                assert_array_equals( _events,
```

Comment:
Was [EventWatcher](https://github.com/w3c/testharness.js/blob/master/docs/api.md) considered for checking event sequence?  Probably because of a few noisy events it can’t be converted right away, but it is still worth consideration.